### DISCUSSION TITLE
clarify config parameter "server"

### DISCUSSION THREAD

**Discussion by bovy89** [rancher/rke2#3701] [Q&A] [answered]
Can you clarify the role of parameter "server"? Why it is needed is clear, but I need some additional details.

For example, in a setup made of 3 servers + X agents + load balancer:

on agent nodes and server 2 and 3

```
$ cat /etc/rancher/rke2/config.yaml
server: https://<vip_on_loadbalancer>:9345
<other configs>
```

on server 1:

```
$ cat /etc/rancher/rke2/config.yaml
<other configs>
```

Questions:

1) In the above setup, do we need to change server 1 config after cluster init as follows (add server parameter)?

```
$ cat /etc/rancher/rke2/config.yaml
server: https://<vip_on_loadbalancer>:9345
<other configs>
```

2)  If point 1 is true, do we need it also in a single server node setup or is it useless?

3) If we forgot to add `server: https://<vip_on_loadbalancer>:9345` on server 1 after cluster init what can happen? Is it unsafe?

4) If point 1 is true and we want to automatically do that change using a configuration management tool, is there a way to check if the cluster has already been initialized (e.g.  test -e /var/lib/rancher/rke2/server/db/etcd)?

**Comment by brandond** [comment, marked as answer]
--cluster-init and --server options are only important when initializing or joining servers to the cluster. Once a server is a part of the cluster (etcd database files exist on disk), the flags are no longer important. You can leave them set, or remove them, it's up to you.

**Reply by bovy89** [reply]
cluster-init is not an option anymore right?

In order to answer my questions:

1) no, is not needed
2) not necessary because 1 is false. BTW is useless
3) nothing: is completely safe
4) not necessary because 1 is false

@brandond can you confirm that?

**Reply by brandond** [reply]
You're right in that cluster-init is hardcoded internally on RKE2.

Once a server is in the cluster, the addresses of other servers are discovered using the etcd datastore; there is no significant reason to keep the server address in the configuration, other than in case you at some point need to issue a cluster-reset as part of a restore operation, and will need to remove the datastore files from disk on nodes and rejoin them to the cluster again.

**Comment by bovy89** [comment]
@brandond due to the previous answer, I think that the comment here https://github.com/rancher/rke2/discussions/2761#discussioncomment-4122471 is wrong

**Reply by brandond** [reply]
Some users like @Martin-Weiss may prefer to add the server setting to all the servers rather than remove it or simply ignore it. Either way is fine; it doesn't make a huge difference.

**Reply by Martin-Weiss** [reply]
The server: entry is also useful after a join in some cases regarding trouble-shooting or fixing a broken machine.. 
(i.e rke2-unistall.sh and the systemctl start rke2-agent / rke2-server to quickly "repair" a broken agent or server..)
